Fix doc install RHEL from RPM - crb repo do not exist in RHEL

luisarizmendi opened the pull request:
## Description

"dnf config-manager --set-enabled crb" do not work since the crb repo do not exist in any RHEL reelase, that is valid for CentOS Stream only.

### Did you use Generative AI?

No

### Additional Information

https://docs.fedoraproject.org/en-US/epel/getting-started/

luisarizmendi committed:
Fix doc RHEL - cbc repo do not exist in RHEL

cottsay commented on source/Installation/RHEL-Install-RPMs.rst:
```diff
@@ -34,11 +34,11 @@ You will need to enable the EPEL repositories and the PowerTools repository:
-   $ sudo dnf install 'dnf-command(config-manager)' epel-release -y
-   $ sudo dnf config-manager --set-enabled crb
+   $ sudo dnf install -y 'dnf-command(config-manager)' https://dl.fedoraproject.org/pub/epel/epel-release-latest-$(rpm -E %rhel).noarch.rpm
+   $ sudo dnf config-manager --set-enabled codeready-builder-for-rhel-$(rpm -E %rhel)-$(arch)-rpms
```
```suggestion
   $ sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-$(rpm -E %rhel).noarch.rpm
   $ sudo crb enable
```

cottsay reviewed (commented):
Thanks for pointing this out.

When this was written we assumed that most users would be using a RHEL clone like CentOS, AlmaLinux, or Rocky Linux, but it's possible to write instructions that can be used on RHEL or a clone.

In particular, `epel-release` now provides an executable `/usr/bin/crb` that can be used to detect and enable the proper repository. We should use that - it is expected to work the same on RHEL and clones.

fujitatomoya reviewed (approved):
lgtm with resolving @cottsay comments.

luisarizmendi committed:
Update source/Installation/RHEL-Install-RPMs.rst

Ok, but the "crb enable" does not work in some workflows, for example I found problems while creating a bootc image, that's why I stick to the command that always work in RHEL. Maybe we should include commands for RHEL and other tab for RHEL-variants

Co-authored-by: Scott K Logan <logans@cottsay.net>
Signed-off-by: Luis Javier Arizmendi Alonso <luis.ariz@gmail.com>

ahcorde commented:
Do you mind to re-review this PR @cottsay ?

cottsay commented:
I just spotted this in a commit message - I believe it was meant to be a topic of conversation:

> Ok, but the "crb enable" does not work in some workflows, for example I found problems while creating a bootc image, that's why I stick to the command that always work in RHEL. Maybe we should include commands for RHEL and other tab for RHEL-variants

Is it possible that you were trying to run `crb enable` without having installed `epel-release` first? It's hard to help you debug without more information.

luisarizmendi commented:
@cottsay , let me show what I mean by sharing an specific example.

I have a [repo that builds bootc images using GitHub Actions](https://github.com/luisarizmendi/bootc-images). I already have my [bootc ROS image](https://github.com/luisarizmendi/bootc-images/pkgs/container/bootc-rhel-ros) and it's associated ISO ([embeded in another container image](https://github.com/luisarizmendi/bootc-images/pkgs/container/bootc-rhel-ros-anaconda-iso)) based on RHEL 9.6 here, it [was built with this Containerfile](https://github.com/luisarizmendi/bootc-images/blob/8a740d8b6ce9f6672a34878ff8279c38b2105684/rhel-ros/Containerfile).

If I enable crb with the `crb enable` command instead of the specific dnf command ([you can check the change in this commit](https://github.com/luisarizmendi/bootc-images/commit/890ce26c086bb8e43fc0a72aa51d1748ee217813)), you can see how [the built fails](https://github.com/luisarizmendi/bootc-images/actions/runs/18014587080/job/51256126173) when trying to install the `ros-kilted-desktop`...because crb was not enabled:

```
STEP 8/15: RUN dnf install -y --releasever=9 --installroot=/tmp/ros-install ros-kilted-desktop  &&     dnf clean all
Updating Subscription Management repositories.
subscription-manager is operating in container mode.
Extra Packages for Enterprise Linux 9 - x86_64   27 MB/s |  20 MB     00:00    
Extra Packages for Enterprise Linux 9 openh264  3.0 kB/s | 2.5 kB     00:00    
Red Hat Enterprise Linux 9 for x86_64 - BaseOS   70 MB/s |  79 MB     00:01    
Red Hat Enterprise Linux 9 for x86_64 - AppStre  62 MB/s |  69 MB     00:01    
ROS 2 - x86_64                                  2.7 MB/s | 4.7 MB     00:01    
Error: 
 Problem: package ros-kilted-desktop-0.12.0-2.el9.20250906.002012.x86_64 from ros2 requires ros-kilted-pcl-conversions, but none of the providers can be installed
  - conflicting requests
  - nothing provides eigen3-devel needed by ros-kilted-pcl-conversions-2.7.3-1.el9.20250823.034029.x86_64 from ros2
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
Error: building at STEP "RUN dnf install -y --releasever=9 --installroot=/tmp/ros-install ros-kilted-desktop  &&     dnf clean all": while running runtime: exit status 1
Error: Error: buildah exited with code 1
```

If you take a closer look at output of the step where it should be enabled you can see a message stating that the actual repo is disabled, so that it "cannot" be enabled :

```
STEP 4/15: RUN crb enable &&     dnf clean all
Enabling CRB repo
subscription-manager is operating in container mode. Use your host system to manage subscriptions.
CRB repo is disabled
```

cottsay commented:
Thanks for the details, @luisarizmendi. That's unfortunate.

It looks like the `crb` command recently added the `FORCE_DNF` flag, and that appears to do what we want and always runs the dnf command to enable repo. Do you think using `env FORCE_DNF=1 crb enable` is acceptable?

Aside: directly interacting with the dnf config bypasses the subscription manager, which is what the `crb` command is trying to do. Is that a problem?

luisarizmendi commented:
Thanks for the pointer.

Including that variable the [build succeed.](https://github.com/luisarizmendi/bootc-images/actions/runs/18033982695/job/51316620802)

There is no problem enabling repositories using DNF instead subscription-manager, the Red Hat subscriptions are used from the mounted entitlements from the host.

I plan to include information in the ROS doc (future PR) on how to build a ROS bootc image with both Fedora and RHEL. I will include this useful info, thanks.

cottsay commented on source/Installation/RHEL-Install-RPMs.rst:
```diff
@@ -34,11 +34,11 @@ You will need to enable the EPEL repositories and the PowerTools repository:
-   $ sudo dnf install 'dnf-command(config-manager)' epel-release -y
-   $ sudo dnf config-manager --set-enabled crb
+   $ sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-$(rpm -E %rhel).noarch.rpm
+   $ sudo crb enable
```
```suggestion
   $ sudo env FORCE_DNF=1 crb enable
```

cottsay reviewed (commented):
> Including that variable the build succeed.

Cool, let's go with that then. Thanks for iterating with me to arrive at a solution!

cottsay committed:
Use FORCE_DNF=1

Signed-off-by: Scott K Logan <logans@cottsay.net>

cottsay commented:
@Mergifyio backport kilted jazzy humble
